### PR TITLE
Bug 1564820 - Document payload format for all metric types

### PR DESCRIPTION
**Pull request by badboy**
Oh, and also the other bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1566854

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: docs/dev/core/internal/payload.md [hunk 1]
```diff
@@ -11,7 +11,111 @@ It is written as a set of [templates](https://github.com/mozilla-services/mozill
...
+    "values": {
+        "0": 1,
+        "1": 3,
+    }
+}
+```
+
+### UUID
+
+A [UUID](../../../user/metrics/uuid.md) is represented by the string representation of the UUID
```
Comment (mdboom) [inline docs/dev/core/internal/payload.md]:
```suggestion
A [UUID](../../../user/metrics/uuid.md) is represented by the string representation of the UUID.
```

File: docs/dev/core/internal/payload.md [hunk 2]
```diff
@@ -11,7 +11,111 @@ It is written as a set of [templates](https://github.com/mozilla-services/mozill
...
+#### Example
+
+```json
+"29711dc8-a954-11e9-898a-eb4ea7e8fd3f"
+```
+
+### Datetime
+
+A [Datetime](../../../user/metrics/datetime.md) is represented by its ISO8601 string representation, truncated to the metric's time unit.
+It always includes the timezone offset
```
Comment (mdboom) [inline docs/dev/core/internal/payload.md]:
```suggestion
It always includes the timezone offset.
```

File: docs/dev/internal/payload-format.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "range": [0, 60000],
+    "histogram_type": "expontential",
+    "sum": 3,
+    "time_unit": "milliseconds",
+    "values": {
+        "0": 1,
+        "1": 3,
+    }
+}
+```
```
Comment (mdboom) [inline docs/dev/internal/payload-format.md]: This file was maybe included by accident?

Comment (badboy) [reply docs/dev/internal/payload-format.md]: It was...

File: docs/dev/core/internal/payload.md [hunk 4]
```diff
@@ -11,7 +11,111 @@ It is written as a set of [templates](https://github.com/mozilla-services/mozill
...
+
+A [Timing distribution](../../../user/metrics/timing_distribution.md) is represented as an object with the following fields.
+
+| Field name | Type | Description |
+|---|---|---|
+| `bucket_count` | Integer | The bucket count of the histogram. |
+| `range` | Array&lt;Integer&gt; | The range indicated by its minimum and maxium value. |
+| `sum` | Integer | The sum of all recorded values. |
+| `time_unit` | String | The timespan's time unit, see the [time distribution's configuration](../../../user/metrics/timing_distribution.md#configuration) for valid values. |
+| `values` | Map&lt;String, Integer&gt; | The values in each bucket. The key is the bucket's minimum value. Empty buckets are removed. |
```
Comment (Dexterp37) [inline docs/dev/core/internal/payload.md]: Mh, this still gets me a bit confused after having removed the context. What do you think about something like this: "The values in each bucket. The key is the minimum value for the range of that bucket. Buckets with no values are not reported"

**Pull request merged**